packages/legacy_material/lib/src/theme_data.dart

```diff
@@ -0,0 +1,1841 @@
+    dialogBackgroundColor ??= isDark ? Colors.grey[800]! : Colors.white;
+    indicatorColor ??= colorScheme.secondary == primaryColor
+        ? Colors.white
+        : colorScheme.secondary;
+    hintColor ??= isDark ? Colors.white60 : Colors.black.withOpacity(0.6);
+    // The default [buttonTheme] is here because it doesn't use the defaults for
+    // [disabledColor], [highlightColor], and [splashColor].
+    buttonTheme ??= ButtonThemeData(
+      colorScheme: colorScheme,
+      buttonColor: isDark ? primarySwatch[600]! : Colors.grey[300]!,
+      disabledColor: disabledColor,
+      focusColor: focusColor,
+      hoverColor: hoverColor,
+      highlightColor: highlightColor,
+      splashColor: splashColor,
+      materialTapTargetSize: materialTapTargetSize,
+    );
+    disabledColor ??= isDark ? Colors.white38 : Colors.black38;
+    highlightColor ??= isDark
+        ? LegacyMaterialConstants.kDarkThemeHighlightColor
+        : LegacyMaterialConstants.kLightThemeHighlightColor;
+    splashColor ??= isDark
+        ? LegacyMaterialConstants.kDarkThemeSplashColor
+        : LegacyMaterialConstants.kLightThemeSplashColor;
+
+    // TYPOGRAPHY & ICONOGRAPHY
+    typography ??= Typography.material2014(platform: platform);
+    TextTheme defaultTextTheme = isDark ? typography.white : typography.black;
+    TextTheme defaultPrimaryTextTheme =
+        primaryIsDark ? typography.white : typography.black;
+    if (fontFamily != null) {
+      defaultTextTheme = defaultTextTheme.apply(fontFamily: fontFamily);
+      defaultPrimaryTextTheme = defaultPrimaryTextTheme.apply(
+        fontFamily: fontFamily,
+      );
+    }
+    if (fontFamilyFallback != null) {
+      defaultTextTheme = defaultTextTheme.apply(
+        fontFamilyFallback: fontFamilyFallback,
+      );
+      defaultPrimaryTextTheme = defaultPrimaryTextTheme.apply(
+        fontFamilyFallback: fontFamilyFallback,
+      );
+    }
+    if (package != null) {
+      defaultTextTheme = defaultTextTheme.apply(package: package);
+      defaultPrimaryTextTheme = defaultPrimaryTextTheme.apply(package: package);
+    }
+    textTheme = defaultTextTheme.merge(textTheme);
+    primaryTextTheme = defaultPrimaryTextTheme.merge(primaryTextTheme);
+    iconTheme ??= isDark
+        ? IconThemeData(color: kDefaultIconLightColor)
+        : IconThemeData(color: kDefaultIconDarkColor);
+    primaryIconTheme ??= primaryIsDark
+        ? const IconThemeData(color: Colors.white)
+        : const IconThemeData(color: Colors.black);
+
+    // COMPONENT THEMES
+    // TODO(Piinks): We may need to create custom component themes as we go
+    //  forward here to get the right styling. For example, we may need to
+    //  create LegacyAppBarTheme, that includes more properties than the default
+    //  theme provides. The default themes could be expanded upon as well, but
+    //  should be done separately so that development of this package is not
+    //  stalled by having to wait for stable releases.
```

Comment:
Just a question here. If we have new properties in `LegacyXTheme` instead of `XTheme`, how would these properties be applied to the widgets? In the widget implementation, should we also get something like `legacyTheme = LegacyXTheme.of(context)` to get the widget style?

Reply:
Thank you for catching this! This is an older comment I meant to remove, because you are very right - adding new properties to `LegacyXThemeData` wouldn't mean the widget `X` would adopt them.

> If we have new properties in LegacyXTheme instead of XTheme, how would these properties be applied to the widgets?

In order to do this right, we'll need to expand all component themes to include all of the properties we need, and then have the widgets adopt all of the properties from the theme if it is provided. 

 > In the widget implementation, should we also get something like legacyTheme = LegacyXTheme.of(context) to get the widget style?

No, because the component theme should come from Theme.of(context). Right now component theme look ups fallback on the primary theme:

https://github.com/flutter/flutter/blob/1e6a9024738787f862db8991ed325efead5b09d0/packages/flutter/lib/src/material/switch_theme.dart#L223-L226

We wouldn't be able to override a static `.of(context)` of a component theme in a way that would work for the widget, and we don't want the framework to depend on this package. What I envision right now is, in `LegacyThemeData` it will create `LegacyXThemeData` for widget `X`, which instead of having null property values, it will supply the default values from the M2Defaults we have scattered all over the framework. So when we call on `Theme.of(context).xTheme` we get the legacy theme data.

There would definitely be limitations; we are trying our best to support an outdated spec here. But I think it will be the best experience we can provide.

Reply:
Before moving forward with this, I am working on a doc that will lay it all out. :)